libff/algebra/fields/field.hpp

```diff
@@ -0,0 +1,101 @@
+/** @file
+ *****************************************************************************
+ Declaration of common API for all finite fields.
+
+ Currently NOT used by the fields in this library. This class is not actually
+ the parent class of any field. All APIs are enforced through tests instead.
+
+ The reason for this is to ensure high performance of all fields. This class
+ exists as documentation for common API between fields.
+
+ Includes two types of fields, F[p^n] for selected n and F[2^n] for a separate
```

Comment:
Nice! Didn't realize this API would support both prime and binary field extensions